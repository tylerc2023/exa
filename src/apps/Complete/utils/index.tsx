export * from './configs';
export * from './useKeyboardHeight';
export * from './userFactory';
