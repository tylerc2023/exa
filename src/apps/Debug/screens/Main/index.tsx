import React, { memo } from "react";
import { Button, Screen } from "../../../../components";
import { NavigationScreen } from "../../../../models";
import { useNav } from "../../../../hooks";

const debugScreens: NavigationScreen[] = [
  "debugArticle",
  "debugBall",
  "debugButtons",
  "debugCards",
  "debugChat",
  "debugDrag",
  "debugFonts",
  "debugImageCollection",
  "debugInput",
  "debugOKRs",
  "debugPinchSpread",
  "debugQuestionnaire",
  "debugSearchBar",
  "debugStopwatch",
  "debugSwipeCell",
  "debugDarkMode"
];

export default memo(function Debug() {
  const nav = useNav();
  return (
    <Screen onLeftPress={nav.to("portfolioLanding")} title="Debug">
      {debugScreens.map(screen => 
        <Button title={screen} key={screen} onPress={nav.to(screen)} />
      )}
    </Screen>
  );
});
