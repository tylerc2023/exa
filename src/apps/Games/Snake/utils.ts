export type BoardObject = Array<Array<0 | 1 | 2>>;

export const generateRandom = (size: number) =>
  Math.floor(Math.random() * size);

export const addFood = (
  board: BoardObject,
  used: {[key: string]: boolean} = {},
): boolean => {
  const x = generateRandom(board.length);
  const y = generateRandom(board.length);

  const combo = `${x}${y}`;
  if (Object.keys(used).length === board.length) {
    return false;
  }
  if (combo in used || board[x][y] !== 0) {
    used[combo] = true;
    addFood(board, used);
    return false;
  }
  board[y][x] = 2;
  return true;
};

export const addStarting = (board: BoardObject) => {
  const center = Math.floor(board.length / 2);
  board[center][center] = 1;
};

export const generateBoard = (size: number) => {
  const board: BoardObject = [];
  for (let i = 0; i < size; i++) {
    board[i] = [];
    for (let j = 0; j < size; j++) {
      board[i][j] = 0;
    }
  }

  addStarting(board);
  addFood(board);
  return board;
};

export const nextSnakePosition = () => undefined;
export const collision = (next: any) => next && false;
export const endGame = () => undefined;
export const eat = () => undefined;
