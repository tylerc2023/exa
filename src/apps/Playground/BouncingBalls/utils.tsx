import {Animated} from 'react-native';
import {State} from './BouncingBall';
import {CanvasDimensions} from './Canvas';

const getRandomNumber = (min: number, max: number): number =>
  Math.random() * (max - min) + min;

type GetSpeedProps = {
  state: State;
  canvas: CanvasDimensions;
  diameter: number;
  pulse: boolean;
};

export const getSpeed = ({state, canvas, diameter, pulse}: GetSpeedProps) => {
  const dMax = state.diameter >= diameter * 1.1;
  const dMin = state.diameter <= diameter * 0.9;
  // TODO: recenter so pulsing happens from middle
  const growing = dMax || dMin ? !state.growing : state.growing;
  const d = state.diameter + (!pulse ? 0 : growing ? 1 : -1);
  const xMin = canvas.x;
  const xMax = canvas.x + canvas.width;
  const yMin = canvas.y;
  const yMax = canvas.y + canvas.height;
  const xLarger = state.x + d >= xMax;
  const xSmaller = state.x <= xMin;
  const yLarger = state.y + d >= yMax;
  const ySmaller = state.y <= yMin;
  // TODO: add collision
  const xSpeed = xLarger || xSmaller ? -state.xSpeed : state.xSpeed;
  const ySpeed = yLarger || ySmaller ? -state.ySpeed : state.ySpeed;
  const x = xLarger ? xMax - d : xSmaller ? xMin : state.x + xSpeed;
  const y = yLarger ? yMax - d : ySmaller ? yMin : state.y + ySpeed;
  const dx = x + xSpeed;
  const dy = y + ySpeed;
  return {xSpeed, ySpeed, x: dx, y: dy, d, growing};
};

type GetInitialStateProps = {
  canvas: CanvasDimensions;
  diameter: number;
  speed: number;
};

export const getInitialState = ({
  canvas,
  diameter,
  speed,
}: GetInitialStateProps): State => {
  const starting = {
    x: getRandomNumber(canvas.x, canvas.x + canvas.width - diameter),
    y: getRandomNumber(canvas.y, canvas.y + canvas.height - diameter),
  };
  const growing = !!Math.floor(Math.random());
  return {
    ...starting,
    position: new Animated.ValueXY(starting),
    xSpeed: getRandomNumber(-speed, speed),
    ySpeed: getRandomNumber(-speed, speed),
    diameter,
    growing,
  };
};
