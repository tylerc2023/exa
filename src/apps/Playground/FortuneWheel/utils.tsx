type GetRandomRange = {
  min: number;
  max: number;
};

const getRandomRange = ({min, max}: GetRandomRange) => {
  return Math.floor(Math.random() * (max - min) + min);
};

type GetNewLocation = {
  minSpin: number;
  maxSpin: number;
  numOfSegments: number;
  location: number;
};

export const getNewLocation = ({
  minSpin,
  maxSpin,
  numOfSegments,
  location,
}: GetNewLocation) => {
  const spins = getRandomRange({min: minSpin, max: maxSpin}) * 360;
  const winner = Math.floor(Math.random() * numOfSegments);
  const winnerSection = Math.floor(winner * (360 / numOfSegments));
  const newLocation = location + spins + winnerSection;
  return newLocation;
};

type GetWinnerIndex = {
  location: number;
  numOfSegments: number;
  angleOfSegment: number;
};

export const getWinnerIndex = ({
  location,
  numOfSegments,
  angleOfSegment,
}: GetWinnerIndex) => {
  const deg = Math.abs(Math.round(location % 360));
  const index =
    (numOfSegments - Math.floor(deg / angleOfSegment)) % numOfSegments;
  return index;
};
