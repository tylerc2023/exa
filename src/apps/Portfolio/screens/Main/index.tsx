import * as React from "react";
import { connect } from "react-redux";
import { Button, Screen, Text } from "../../../../components";
import { navigate, NavigationScreen } from "../../../../models";

interface DispatchProps {
  navigate: typeof navigate;
}

type Props = DispatchProps;

class Component extends React.PureComponent<Props> {
  public render() {
    return (
      <Screen>
        <Text h1 center title="Main" />
        <Button
          title="logout"
          onPress={this.nav(NavigationScreen.PortfolioLanding)}
        />
      </Screen>
    );
  }
  private nav = (to: NavigationScreen) => () => this.props.navigate(to);
}

const mapDispatchToProps: DispatchProps = { navigate };

export const Main = connect(
  null,
  mapDispatchToProps
)(Component);
