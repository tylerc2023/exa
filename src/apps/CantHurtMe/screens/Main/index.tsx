import React, { memo, useState } from 'react';
import { FlatList, View } from 'react-native';
import { Card, Dialog, Screen, Icon } from '../../../../components';
import { getLandscapeOrientation } from '../../../../models';
import { Theme, useRootSelector } from '../../../../utils';
import { DailyProgress, Header, ProfileLevel } from '../../components';
import { app } from '../../data';
import { useNav } from '../../../../hooks';
import { Item } from './Item';

const initialState = { settings: false, profile: false };

export default memo(function CantHurtMeMain() {
  const [showModal, setShowModal] = useState(initialState);
  const landscape = useRootSelector(state => getLandscapeOrientation(state));
  const columns = landscape ? 4 : 2;
  const nav = useNav();
  const keyExtractor = (id: string) => app.goals.byId[id].id;

  const handleModalBackgroundPress = () => setShowModal(initialState);
  const handleProfilePress = () =>
    setShowModal({ ...initialState, profile: true });
  const handleSettingsPress = () =>
    setShowModal({ ...initialState, settings: true });

  return (
    <>
      <Screen
        border
        onLeftPress={nav.to('portfolioLanding')}
        title="Can't Hurt Me"
      >
        <View style={{ marginHorizontal: Theme.padding.p04 }}>
          <FlatList
            keyExtractor={keyExtractor}
            key={columns}
            data={app.goals.orderById}
            renderItem={Item}
            numColumns={columns}
            ListHeaderComponent={
              <View>
                <Header title="Progress" />
                <Card>
                  <View
                    style={{
                      flexDirection: 'row',
                      justifyContent: 'space-between',
                    }}
                  >
                    <ProfileLevel onPress={handleProfilePress} />
                    <Icon name="settings" onPress={handleSettingsPress} />
                  </View>
                  <DailyProgress />
                </Card>
                <Header title="Challenges" />
              </View>
            }
            ListFooterComponent={
              <View style={{ paddingBottom: Theme.padding.p04 }} />
            }
          />
        </View>
      </Screen>
      {showModal.profile && (
        <Dialog
          duration={2000}
          title="profile"
          onBackgroundPress={handleModalBackgroundPress}
        />
      )}
      {showModal.settings && (
        <Dialog
          title="settings"
          onBackgroundPress={handleModalBackgroundPress}
        />
      )}
    </>
  );
});
