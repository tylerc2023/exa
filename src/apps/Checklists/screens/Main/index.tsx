import React, { memo, useCallback } from "react";
import { FlatList } from "react-native";
import { Button, Screen } from "../../../../components";
import { useNav, useColor } from "../../../../hooks";
import { useRootSelector, useRootDispatch } from "../../../../utils";
import {
  setActiveList,
  getActiveChecklistOrderByCreatedAt
} from "../../models";
import { navigate } from "../../../../models";

export default memo(function Checklists() {
  const nav = useNav();
  const color = useColor();
  const dispatch = useRootDispatch();
  const items = useRootSelector(getActiveChecklistOrderByCreatedAt);

  const handleItemPress = (id: string) => () => {
    dispatch(setActiveList(id));
    dispatch(navigate("checklistsList"));
  };

  const handleItemLongPress = (id: string) => () => {
    dispatch(setActiveList(id));
    dispatch(navigate("checklistsListUpdate"));
  };

  const renderItem = useCallback(
    ({ item }) => 
      <Button
        title={item.name}
        onPress={handleItemPress(item.id)}
        onLongPress={handleItemLongPress(item.id)}
      />
    ,
    [handleItemLongPress, handleItemPress]
  );

  const keyExtractor = useCallback(item => item.id, []);

  return (
    <Screen onLeftPress={nav.to("portfolioLanding")} title="Checklists" gutter>
      <FlatList
        keyExtractor={keyExtractor}
        data={items}
        renderItem={renderItem}
      />
      <Button
        right
        contained
        fab
        icon="plus"
        iconColor={color.background}
        onPress={nav.to("checklistsListCreate")}
      />
    </Screen>
  );
});
