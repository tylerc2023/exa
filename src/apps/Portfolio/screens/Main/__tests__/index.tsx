import * as React from "react";
import { mockRenderer } from "../../../../../utils/Mock";
import { Main } from "../../Main";

it("renders correctly", () => {
  const dom = mockRenderer(<Main />).toJSON();
  expect(dom).toMatchSnapshot();
});
