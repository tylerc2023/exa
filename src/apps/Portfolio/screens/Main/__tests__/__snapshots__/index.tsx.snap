// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "hsl(0, 0%, 100%)",
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "hsl(0, 0%, 100%)",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "height": 72,
          "justifyContent": "space-between",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 0.2,
            "padding": 12,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "hsla(0, 0%, 0%, 1)",
            "fontSize": 40,
            "fontWeight": "100",
            "letterSpacing": -0.5,
            "opacity": 1,
          }
        }
      >
        Main
      </Text>
      <View
        style={
          Object {
            "flex": 0.2,
            "padding": 12,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "hsl(0, 0%, 100%)",
            "flex": 1,
          },
          Object {
            "padding": 16,
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "padding": 8,
            "paddingHorizontal": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "hsl(0, 0%, 0%)",
              "fontSize": 14,
              "fontWeight": "600",
              "letterSpacing": 0.75,
              "opacity": 1,
              "textAlign": "center",
            }
          }
        >
          LOGOUT
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "right": 0,
        },
        Object {
          "height": 0,
        },
        undefined,
      ]
    }
  />
</View>
`;
